main.go

```diff
@@ -76,8 +76,8 @@ func pipe(r io.Reader, w io.WriteCloser) {
 	fmt.Println("Closing pipe")
 }
 
-// pass a stopChan to the go routtine
-func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan chan struct{}) {
+// pass a stopChan to the go routine
+func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan *chan struct{}) {
```

Comment:
My understanding is that we're mutating the state of it in the `master` function. but I will say that I'm only beginning to understand simple pointers. I think I fixed the proxy in 78d72bbd4d0d4a3f65edb406453c2aada45c5e10